tests/test_file_writer.py

```diff
@@ -22,3 +43,206 @@ def test_MantarrayH5FileCreator__sets_file_name_and_userblock_size_and_file_vers
+        assert reference_data[0][0] == 20
+        assert reference_data[0][-1] == 1182980
+        assert reference_data[1][0] == -2722104
+        assert reference_data[1][-1] == -4190110
+
+        wf.get_h5_file().close()  # safe clean-up when running CI on windows systems
+
+
+def test_h5_file_trimmer__When_invoked_on_a_current_file_with_args_on_time_points__Then_the_new_file_has_trimmed_raw_referene_and_tissue_data(
+    current_version_file_path,
+):
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        new_file_path = h5_file_trimmer(current_version_file_path, tmp_dir, 320, 320)
+
+        wf = WellFile(new_file_path)
+
+        assert wf.get_h5_attribute(str(TRIMMED_TIME_FROM_ORIGINAL_START_UUID)) == 320
+        assert wf.get_h5_attribute(str(TRIMMED_TIME_FROM_ORIGINAL_END_UUID)) == 320
+
+        # raw data
+        tissue_data = wf.get_raw_tissue_reading()
+        assert tissue_data[0][0] == 440
+        assert tissue_data[0][-1] == 135640
+        assert tissue_data[1][0] == -950718
+        assert tissue_data[1][-1] == 1713713
+
+        reference_data = wf.get_raw_reference_reading()
+        assert reference_data[0][0] == 340
+        assert reference_data[0][-1] == 1182660
+        assert reference_data[1][0] == -2654995
+        assert reference_data[1][-1] == -4123001
+
+        wf.get_h5_file().close()  # safe clean-up when running CI on windows systems
+
+
+def test_h5_file_trimmer__When_invoked_on_a_current_file_with_only_end_arg__Then_the_new_file_has_trimmed_raw_referene_and_tissue_data(
+    current_version_file_path,
+):
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        new_file_path = h5_file_trimmer(
+            current_version_file_path, tmp_dir, from_end=320
+        )
+
+        wf = WellFile(new_file_path)
+
+        assert wf.get_h5_attribute(str(TRIMMED_TIME_FROM_ORIGINAL_START_UUID)) == 0
+        assert wf.get_h5_attribute(str(TRIMMED_TIME_FROM_ORIGINAL_END_UUID)) == 320
+
+        # raw data
+        tissue_data = wf.get_raw_tissue_reading()
+        assert tissue_data[0][0] == 120
+        assert tissue_data[0][-1] == 135640
+        assert tissue_data[1][0] == -974940
+        assert tissue_data[1][-1] == 1713713
+
+        reference_data = wf.get_raw_reference_reading()
+        assert reference_data[0][0] == 20
+        assert reference_data[0][-1] == 1182660
+        assert reference_data[1][0] == -2722104
+        assert reference_data[1][-1] == -4123001
+
+        wf.get_h5_file().close()  # safe clean-up when running CI on windows systems
+
+
+def test_h5_file_trimmer__When_invoked_on_a_trimmed_file__Then_the_new_file_is_additionally_trimmed_with_the_raw_reference_tissue_data_and_metadata_updated(
+    trimmed_file_path,
+    mocker,
+):
+
+    mocked_print = mocker.patch("builtins.print", autospec=True)
+
+    mocked_start_str = "160 centimilliseconds was trimmed from the start instead of 200"
+    mocked_end_str = "160 centimilliseconds was trimmed from the end instead of 200"
```

Comment:
note for future: when asserting things about messages, you can use `in` to focus on the key portions so that minor changes to grammar in the future wouldn't break the test.  for example `assert '160 centimilliseconds' in mocked_print.callargs...` (you can use multiple `in` or a regular expression match if multiple portions of the message are important)